### PR TITLE
Add inlinehelp button to FormView

### PR DESCRIPTION
**Pull request by HLeithner**
Based on #37158 adding this functionallity to the FormView

### Summary of Changes
Add inlinehelp button to FormView

### Testing Instructions
Take a component using FormView and add the xml to the form.xml
```xml
  <config>
    <inlinehelp button="show"/>
  </config>
```
Watch out for the button

### Actual result BEFORE applying this Pull Request
Nothing happens

### Expected result AFTER applying this Pull Request
Shows the inline help button

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [X] Pull Request link for manual.joomla.org: TBA
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/MVC/View/FormView.php [hunk 1]
```diff
@@ -230,6 +231,15 @@ protected function addToolbar()
 
         ToolbarHelper::divider();
 
+        if ($this->form instanceof Form) {
+            $formConfig  = $this->form->getXml()->config->inlinehelp;
+            $targetClass = (string)$formConfig['targetclass'] ?: 'hide-aware-inline-help';
```
Comment (heelc29) [inline libraries/src/MVC/View/FormView.php]:
```suggestion
            $targetClass = (string) $formConfig['targetclass'] ?: 'hide-aware-inline-help';
```

File: libraries/src/MVC/View/FormView.php [hunk 2]
```diff
@@ -230,6 +231,15 @@ protected function addToolbar()
...
+
+            if ((string)$formConfig['button'] === 'show') {
```
Comment (heelc29) [inline libraries/src/MVC/View/FormView.php]:
```suggestion
            if ((string) $formConfig['button'] === 'show') {
```

**Review by heelc29 — COMMENTED** [review]
cs

File: libraries/src/MVC/View/FormView.php [hunk 3]
```diff
@@ -230,6 +231,15 @@ protected function addToolbar()
 
         ToolbarHelper::divider();
 
+        if ($this->form instanceof Form) {
+            $formConfig  = $this->form->getXml()->config->inlinehelp;
```
Comment (wilsonge) [inline libraries/src/MVC/View/FormView.php:243]: We probably need to check there is an inlinehelp property here for 3rd party extensions

Comment (HLeithner) [reply libraries/src/MVC/View/FormView.php:243]: that's the only positive thing on simple_xml objects you always get an simple_xml_object back:
https://3v4l.org/W0dcn

so no we don't need to check if the property exists

```php
$xmlstring = <<<xml
<form>
  <config>
    <inlinehelp button="show"/>
  </config>
</form>
xml;

$xml = simplexml_load_string($xmlstring);

$inlinehelp  = $xml->config->inlinehelp;
$nohelp  = $xml->config->nohelp;

var_dump($inlinehelp);
var_dump($nohelp);
```

**Review by richard67 — APPROVED** [review]

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.0-dev. No new features will be merged into Joomla! 4.3 series. Joomla! 4.4 series is a bridge release to make migration from Joomla! 4 to 5 as smooth as possible.

**Pull request merged**